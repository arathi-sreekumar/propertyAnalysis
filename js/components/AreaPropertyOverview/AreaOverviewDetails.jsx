// @flow

import React from 'react';
import { string, shape } from 'prop-types';

type FlowAreaData = {
	area: {
		area_name?: string,
		country?: string,
		county?: string,
		postcode?: string,
		street?: string,
		town?: string,
		zed_index?: string
	}
};

/*
  * AreaOverviewDetails component displays basic details regarding an area
*/
const AreaOverviewDetails = (props: FlowAreaData) => {
	const county = props.area.county;
	const postcode = props.area.postcode;
	const street = props.area.street;
	const town = props.area.town;
	const country = props.area.country;
	const zedIndex = props.area.zed_index;
	const areaName = props.area.area_name;

	return (
		<ul>
			{areaName && <li>Area: {areaName}</li>}
			<li>Country: {country}</li>
			<li>County: {county}</li>
			{postcode && <li>PostCode: {postcode}</li>}
			{street && <li>Street: {street}</li>}
			{town && <li>Town: {town}</li>}
			<li> Average property price: Â£{zedIndex} </li>
		</ul>
	);
};

AreaOverviewDetails.propTypes = {
	area: shape({
		area_name: string,
		country: string,
		county: string,
		postcode: string,
		street: string,
		town: string,
		zed_index: string
	}).isRequired
};

AreaOverviewDetails.defaultProps = {
	area: {
		area_name: '-',
		country: '-',
		county: '-',
		postcode: '-',
		street: '-',
		town: '-',
		zed_index: '-'
	}
};

export default AreaOverviewDetails;
