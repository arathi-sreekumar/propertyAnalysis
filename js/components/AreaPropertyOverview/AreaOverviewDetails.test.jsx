import React from 'react';
import { shallow } from 'enzyme';
import AreaOverviewDetails from './AreaOverviewDetails';

const AREA_DATA = {
	street: '',
	town: '',
	zed_index: '295212',
	postcode: 'BN10',
	country: 'England',
	area_name: ' BN10',
	county: 'East Sussex'
};

describe('AreaOverviewDetails', () => {
	it('renders correctly', () => {
		const component = shallow(<AreaOverviewDetails area={AREA_DATA} />);
		expect(component).toMatchSnapshot();
	});
});
