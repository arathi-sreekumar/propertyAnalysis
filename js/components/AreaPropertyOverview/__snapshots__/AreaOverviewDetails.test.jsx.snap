// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AreaOverviewDetails renders correctly 1`] = `
<ul>
  <li>
    Area: 
     BN10
  </li>
  <li>
    Country: 
    England
  </li>
  <li>
    County: 
    East Sussex
  </li>
  <li>
    PostCode: 
    BN10
  </li>
  <li>
     Average property price: Â£
    295212
     
  </li>
</ul>
`;
